core/thread_flags.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     core_thread
+ * @{
+ *
+ * @file
+ * @brief       thread flags implementation
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+
+#include "bitarithm.h"
+#include "thread_flags.h"
+#include "irq.h"
+#include "thread.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#ifdef MODULE_CORE_THREAD_FLAGS
+static thread_flags_t _thread_flags_clear_atomic(thread_t *thread, thread_flags_t mask)
+{
+    unsigned state = irq_disable();
+    mask &= thread->flags;
+    thread->flags &= ~mask;
```

Comment:
I am not 100% sure, but I think
```
mask &= thread->flags;
thread->flags &= ~mask;
```
can be reduced to `thread->flags &= ~mask;` only?